File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        auto m = std::unordered_map<int, int>();

        for(int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            auto it = m.find(target-num);
            if(it != m.end()) {
                return vector<int>{it->second, i};
            }
            m[num] = i;
        }
        return vector<int>{-1, -1};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(numbers []int, target int) []int {
    p, q := 0, len(numbers) - 1

    for p < q {
        x := numbers[p] + numbers[q] - target
        if x == 0 {
            return []int{p+1, q+1}
        }
        if x > 0 {
            q -= 1
        } else {
            p += 1
        }
    }
    return []int{0, 0}
}